source/scripts/Background/controllers/account/syscoin.ts

```diff
@@ -99,61 +104,99 @@ const SysAccountController = (): ISysAccountController => {
-                syscoin: account.assets,
-              },
-            };
-          else
-            return {
-              ...account,
-              label: accounts[index].label,
-              assets: accounts[index].assets,
-              transactions: [...filteredTxs],
-            };
-        }
-        if (isBitcoinBased)
-          return {
-            ...account,
-            label: accounts[index].label,
-            transactions: [...filteredTxs],
-            assets: {
-              ethereum: accounts[index].assets.ethereum,
-              syscoin: account.assets,
-            },
-          };
-        else
-          return {
-            ...account,
-            label: accounts[index].label,
-            assets: accounts[index].assets,
-            transactions: [...allTxs],
-          };
-      })
-    );
-    store.dispatch(
-      setAccounts({
-        ...formattedWalletAccountsLatestUpdates,
-      })
+      await Promise.all(
+        Object.values(walleAccountstLatestUpdate).map(
+          async (account: any, index) => {
+            const { transactions: updatedTxs } = account;
+
+            const allTxs = [
+              ...accounts[index].transactions,
+              ...updatedTxs,
+            ].filter(
+              (value, i, self) =>
+                i ===
+                self.findIndex((tx) => tx && value && tx[hash] === value[hash])
+            ); // to get array with unique txs.
+            if (index === accountId) {
+              if (isBitcoinBased)
+                return {
+                  ...account,
+                  label: accounts[index].label,
+                  transactions: [...filteredTxs],
+                  assets: {
+                    ethereum: accounts[index].assets.ethereum,
+                    syscoin: account.assets,
+                  },
+                };
+              else {
+                //UPDATE ETH ERC TOKEN BALANCES
+                const getUpdatedErcTokens = await getBalanceUpdatedToErcTokens(
+                  accounts[index].id
+                );
+
+                return {
+                  ...account,
+                  label: accounts[index].label,
+                  assets: {
+                    syscoin: accounts[index].assets.syscoin,
+                    ethereum: getUpdatedErcTokens,
+                  },
+                  transactions: [...filteredTxs],
+                };
+              }
+            }
+            if (isBitcoinBased)
+              return {
+                ...account,
+                label: accounts[index].label,
+                transactions: [...filteredTxs],
+                assets: {
+                  ethereum: accounts[index].assets.ethereum,
+                  syscoin: account.assets,
+                },
+              };
+            else {
+              //UPDATE ETH ERC TOKEN BALANCES
+              const getUpdatedErcTokens = await getBalanceUpdatedToErcTokens(
```

Comment:
Also is possible to check here the dispatch of setAccounts, which is only triggered on this function is also losing information regarding the Ethereum state is possible to check in the below print:
<img width="1135" alt="image" src="https://user-images.githubusercontent.com/21185537/215211251-7fcc3f3f-c6d5-4836-b41b-85671cdcda43.png">
It needs to be reviewed as well.